common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -693,14 +701,83 @@ public static int hashCodeAscii(byte[] bytes, int startPos, int length) {
      * The resulting hash code will be case insensitive.
      */
     public static int hashCodeAscii(CharSequence bytes) {
-        if (!hasUnsafe() || !unalignedAccess()) {
-            return hashCodeAsciiSafe(bytes);
-        } else if (PlatformDependent0.hasCharArray(bytes)) {
-            return PlatformDependent0.hashCodeAscii(PlatformDependent0.charArray(bytes));
-        } else if (PlatformDependent0.hasByteArray(bytes)) {
-            return PlatformDependent0.hashCodeAscii(PlatformDependent0.byteArray(bytes));
+        int hash = HASH_CODE_ASCII_SEED;
+        final int remainingBytes = bytes.length() & 7;
+        // Benchmarking shows that by just naively looping for inputs 8~31 bytes long we incur a relatively large
+        // performance penalty (only achieve about 60% performance of loop which iterates over each char). So because
+        // of this we take special provisions to unroll the looping for these conditions.
+        switch (bytes.length()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of switch cases from 33 to at most 30. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1479)